libsolidity/codegen/CompilerContext.h

```diff
@@ -37,13 +37,24 @@ namespace dev {
 namespace solidity {
 
 
+/// Depending on the compilation is on the runtime code or the creation code,
+/// the interpretation of internal function values differ.
+enum class CompilationMode { Runtime, Creation };
+
 /**
  * Context to be shared by all units that compile the same contract.
  * It stores the generated bytecode and the position of identifiers in memory and on the stack.
  */
 class CompilerContext
 {
 public:
+	CompilerContext(CompilationMode _mode, CompilerContext* _runtimeContext = nullptr) :
```

Comment:
A less explicit way of doing the same thing would be to use the null-ness of `_runtimeContext` as the switch.